### PR TITLE
Increase FPS for FireFlie to 40

### PR DESCRIPTION
**Pull request by Doctoror**
FPS is increased by generating new frames by linear averaging between existing frames. Example:

Let's say we had animation that starts with following frames:
[0f, 1f, 2f, 3f],
[2f, 3f, 4f, 5f]
[     ...      ]
[..last frame..]

Then, one more frame between each frame pairs is inserted:
[0f, 1f, 2f, 3f],
[1f, 2f, 3f, 4f],
[2f, 3f, 4f, 5f]
[     ...      ]
[..last frame..]

This way, we get `2x - 1` frames, which means that we can increase from 20 fps to 40.

Now, every scene will have its own FPS setting.
Previous scenes will remain at 20 until all of them are improved.

I tried generating for 60 fps, looks good but Emulator cannot pull it off, it starts lagging.
I'm afraid to overload real devices too so decided to stick with 40.

### CODE REVIEW

File: java/com/scrat/everchanging/EverchangingRender.java [hunk 1]
```diff
@@ -208,11 +208,15 @@ public void onSurfaceChanged(final GL10 gl, final int width, final int height) {
 
         final float scaleImageHeight = surfaceHeight / 320f;
         final float scaleImageWidth = 240f / width;
-        //todo setup position
+
         final int scenesSize = scenes.size();
         for (int i = 0; i < scenesSize; i++) {
-            scenes.get(i)
-                    .setupPosition(surfaceWidth, surfaceHeight, (i<scenesSize-1)?scaleImageHeight:scaleImageWidth, displayRotation);
+            scenes.get(i).setupPosition(
```
Comment (Doctoror) [inline java/com/scrat/everchanging/EverchangingRender.java:214]: Just a reformat

File: java/com/scrat/everchanging/FireFlie.java [hunk 2]
```diff
@@ -126,69 +234,96 @@ public class FireFlie extends TextureObject {
...
+
+    private int numClips = minObjects;
+    private int frameCounter = 0;
+    private boolean init = false;
 
     private final Calendar calendar;
 
-    static final String[][] textureList = {{"image_168", "image_170"}};
-    static final float[][] pivotList = {{7.5f, 7.5f}, {7.5f, 7.5f}};
+    private static final String[][] textureList = {{"image_168", "image_170"}};
```
Comment (Doctoror) [inline java/com/scrat/everchanging/FireFlie.java:283]: Seems I missed this class when reformatting code and changing visibility modifiers.
I added similar change in this PR.

**Comment by Doctoror** [comment]
I opened this but now I realized there's a problem when multiple animations are taking place.
The ones that are ending will run in a FPS dictated to a new scene.

**Comment by SCratORS** [comment]
Should I merge it?
I created a release just in case before that.

**Comment by Doctoror** [comment]
Depends on what you think about the problem with phasing out animations.
Maybe we can close this without merging and once I have most of the scenes migrated to 40 FPS then we can merge.

**Comment by SCratORS** [comment]
That's why I created the release before that. Here it is also necessary to take into account that some animations, in addition to simply moving an object, use different textures for each frame of the animation. And if the movement can be split into additional frames, then what to do with textures? For example, "Leaves" or "Petals"

**Comment by Doctoror** [comment]
>  in addition to simply moving an object, use different textures for each frame of the animation. And if the movement can be split into additional frames, then what to do with textures? For example, "Leaves" or "Petals".

Yes, I don't have a solution to this yet. I am working on those that can be improved without textures first.
I was thinking about leaving them at 20 or at least translating them at 40 but keeping the texture change at the same pace (until we figure out how to produce intermediate textures).

**Comment by SCratORS** [comment]
> Another problem that I am facing is "code too large", because of a huge array it cannot compile some classes.

Seriously? I didn't know that.  О_О

Basically, I don't mind starting to increase FPS. And we'll see what happens next. =)

**Pull request closed without merging**